runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -110,7 +107,8 @@ static PackageAttributes createPackageAttributes(File source,
 
       // Create the DataflowPackage with staging name and location.
       String uniqueName = getUniqueContentName(source, hash);
-      String resourcePath = IOChannelUtils.resolve(stagingPath, uniqueName);
+      String resourcePath = FileSystems.matchSingleFileSpec(stagingPath).resourceId()
```

Comment:
I think this should be `matchNewResource`. Staging dir may not exist, for example.